Iterable-iOS-SDK/IterableAPI.m

```diff
@@ -650,6 +650,32 @@ - (void)disableDeviceForAllUsersWithOnSuccess:(OnSuccessHandler)onSuccess onFail
 }
 
 // documented in IterableAPI.h
+- (void)updateUser:(NSDictionary *)dataFields mergeNestedObjects:(BOOL)mergeNestedObjects onSuccess:(OnSuccessHandler)onSuccess onFailure:(OnFailureHandler)onFailure
+{
+    NSDictionary *args;
+    if (dataFields) {
+        NSNumber *mergeNested = [NSNumber numberWithBool:mergeNestedObjects];
```

Comment:
There was a coercion error when trying to assign a BOOL in a collection: Collection element of type 'BOOL' is not an objective-C object. 
https://useyourloaf.com/blog/adding-booleans-to-objective-c-arrays-and-dictionaries/

If there an alternative you can think of let me know.